components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -2543,6 +2543,31 @@ public static NativeArray jsFunction_getTiers(Context cx, Scriptable thisObj, Ob
         return myn;
     }
 
+    public static NativeArray jsFunction_getResourceTiers(Context cx, Scriptable thisObj, Object[] args,
+            Function funObj) {
+        NativeArray myn = new NativeArray(1);
+        APIProvider apiProvider = getAPIProvider(thisObj);
+        try {
+            Set<Tier> tiers = apiProvider.getResTiers();
+            List<Tier> tierList = APIUtil.sortTiers(tiers);
+            int i = 0;
+            if (tiers != null) {
+                for (Tier tier : tierList) {
+                    NativeObject row = new NativeObject();
+                    row.put("tierName", row, tier.getName());
+                    row.put("tierDisplayName", row, tier.getDisplayName());
+                    row.put("tierDescription", row, tier.getDescription() != null ? tier.getDescription() : "");
+                    row.put("defaultTier", row, i == 0);
+                    myn.put(i, myn, row);
+                    i++;
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Its not a good practise to catch generic 'Exception'. If one of the above methods throw its better to mention that in a comment. And its also better to explain why we're logging and continuing the process instead of throwing an error.